### ISSUE TITLE
`Datacube.load` should better support `CRS` object and atemporal products

### ISSUE THREAD

**Issue by benjimin** [opendatacube/datacube-core#807]
Support for atemporal products (such as a static DEM) should be improved, particularly in `Datacube.load` as in this example where the `like` parameter behaves unexpectedly. Also, the same function should fully support the `datacube.utils.geometry.CRS` object, rather than requiring e.g. WKT conversion for some fields and not others.

Obvious use case: extract any limited xarray from a datacube, then seek to combine with DEM.

Setup:
```python
lat, lon, e = -35.09407, 150.80426, 0.01
import datacube
dc = datacube.Datacube()
x = dc.load(product='ls8_nbart_albers', time=('2019-06', '2019-07'), 
            lat=(lat-e,lat+e), lon=(lon-e,lon+e))
```
Problem 1: This should return data:
``` python
>>> dc.load(product='dsm1sv10', like=x)
<xarray.Dataset>
Dimensions:  ()
Data variables:
    *empty*
```
Problem 2: This should completely work:
```python
>>> cr = lambda coord: (coord.data.min(), coord.data.max())
>>> dc.load(product='dsm1sv10', x=cr(x.x), y=cr(x.y), 
...         crs=x.crs, output_crs=x.crs, resolution=(25,-25))
TypeError: unhashable type: 'CRS'
```
The combination of the problems: the following does work and is confusingly cumbersome:
```python
>>> dc.load(product='dsm1sv10', x=cr(x.x), y=cr(x.y), 
...         crs=x.crs.wkt, output_crs=x.crs, resolution=(25,-25))
<xarray.Dataset>
Dimensions:    (time: 1, x: 86, y: 100)
...
Attributes:
    crs:      EPSG:3577
```

**Comment by Kirill888** [comment]
@benjimin open time ranges is what needed in this case. The reason why `like` doesn't work in this case is that

1. It injects `time=()` into a query, because input has time axis
2. We do not have open ended ranges so we can't mark "atemporal"  datasets with `time=(0, +inf)`

Work around for your case is to remove `time` dimension when using `like=` parameter:

```python
>>> dc.load(product='dsm1sv10', like=x.isel(time=0))
```

**Comment by Kirill888** [comment]
Also see #808, when merged you'll be able to do:

```python
>>> dc.load(product='dsm1sv10', like=x.geobox)
```

**Comment by benjimin** [comment]
Yep, query by geobox is helpful. (Had totally forgotten I'd offered a PR for that years ago.)

Still, `crs` argument of `Datacube.load` should tolerate a datacube `CRS` object, don't you think?

**Comment by Kirill888** [comment]
@benjimin I missed that part, can you make a new issue for that, certainly `output_crs=` can/expects CRS object, I assumed same applied to `crs=`

**Comment by Kirill888** [comment]
So the CRS part is fixed by #812. Hopefully #808 helps with atemporal products. And since we already have #183,  I'm going to close this issue.
